include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -1111,6 +1107,14 @@ class concurrent_skip_list {
         return create_node(max_level);
     }
 
+    void delete_head() {
+        node_ptr head = my_head_ptr.load(std::memory_order_relaxed);
+        if (head != nullptr) {
+            delete_node(head);
+        }
+        my_head_ptr.store(nullptr, std::memory_order_relaxed);
```

Comment:
May be move it under the `if` statement to avoid restoring `nullptr` into the head?

Reply:
moved